Ladybird/AppKit/Application/ApplicationDelegate.h

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2023, Tim Flynn <trflynn89@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Optional.h>
+#include <AK/StringView.h>
+#include <AK/URL.h>
+#include <Browser/CookieJar.h>
+#include <LibWeb/HTML/ActivateTab.h>
+
+#import <System/Cocoa.h>
+
+@class Tab;
+@class TabController;
+
+@interface ApplicationDelegate : NSObject <NSApplicationDelegate>
```

Comment:
Do we want these to be Obj-C style interfaces? Or can we use Obj-C++ class syntax for them? I believe(?) that 

```
class ApplicationDelegate : public NSApplicationDelegate
{
// stuff
}
```
is valid?

Reply:
Nope - I get `error: base specifier must name a class` with that.

Reply:
Sad. Ok, might tinker with it when I'm feeling apple-y